### ISSUE TITLE
When using `torch.jit.trace` with `Linear+MaxPool2d+BatchNorm2d`, different results are observed.

### ISSUE THREAD

**Issue by Zoeeeeey** [pytorch/pytorch#145207]
### 🐛 Describe the bug

Hi! I found that the following model gives different results after using `torch.jit.trace`.
Are there any bugs in this process?

```python
import numpy as np
import torch
import torch.nn as nn


class SymbolNet(nn.Module):
    def __init__(self):
        super(SymbolNet, self).__init__()
        self.m3 = nn.Linear(in_features=1, out_features=43, bias=True)
        self.m4 = nn.MaxPool2d(kernel_size=(2, 42), stride=2, padding=0, dilation=1, ceil_mode=False)
        self.m5 = nn.BatchNorm2d(1, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)

    def forward(self, x):
        x = self.m3(x)
        x = self.m4(x)
        x = self.m5(x)
        return x


model = SymbolNet()

inp = np.random.rand(24, 1, 4, 1).astype('float32')

m_out = model(torch.from_numpy(inp).to('cpu'))
m_out = [v.cpu().detach() for v in m_out]  # torch2numpy
m_out = [v.resolve_conj().numpy() if v.is_conj() else v.numpy() for v in m_out]

# Compile the model
opt = torch.jit.trace(model.eval(), torch.from_numpy(inp).to('cpu'))
# Compiled run
opt_out = opt(torch.from_numpy(inp).to('cpu'))
opt_out = [v.cpu().detach() for v in opt_out]
opt_out = [v.resolve_conj().numpy() if v.is_conj() else v.numpy() for v in opt_out]

# Differential testing
for i, (l, r) in enumerate(zip(m_out, opt_out)):
    np.testing.assert_allclose(l, r, rtol=1e-2, atol=1e-3, err_msg=f"Result mismatch @ index {i}")

```

Output:
```python
# AssertionError:
# Not equal to tolerance rtol=0.01, atol=0.001
# Result mismatch @ index 0
# Mismatched elements: 2 / 2 (100%)
# Max absolute difference among violations: 2.5436974
# Max relative difference among violations: 2.408335
#  ACTUAL: array([[[-0.560976],
#         [-1.487492]]], dtype=float32)
#  DESIRED: array([[[1.169456],
#         [1.056206]]], dtype=float32)

```

### Versions

```bash
Collecting environment information...
PyTorch version: 2.7.0.dev20250116+cpu
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.10.16 (main, Dec 11 2024, 16:24:50) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-100-generic-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Ti
Nvidia driver version: 535.104.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.6.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.6.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==2.2.1
[pip3] torch==2.7.0.dev20250116+cpu
[pip3] torchaudio==2.6.0.dev20250116+cpu
[pip3] torchvision==0.22.0.dev20250116+cpu
[conda] numpy                     2.2.1                    pypi_0    pypi
[conda] torch                     2.7.0.dev20250116+cpu          pypi_0    pypi
[conda] torchaudio                2.6.0.dev20250116+cpu          pypi_0    pypi
[conda] torchvision               0.22.0.dev20250116+cpu          pypi_0    pypi
```

**Comment by GodHforever** [comment]
You should init model like this. Otherwise batchnorm wiill update its params in runtime. So it is not a bug...
```
model = SymbolNet().eval()
```
